python/lsst/ctrl/mpexec/preExecInit.py

```diff
@@ -125,19 +124,55 @@ def initializeDatasetTypes(self, graph, registerDatasetTypes=False):
         datasetTypes = PipelineDatasetTypes.fromPipeline(
             pipeline, registry=self.butler.registry, include_configs=True, include_packages=True
         )
-        for datasetType in itertools.chain(
-            datasetTypes.initIntermediates,
-            datasetTypes.initOutputs,
-            datasetTypes.intermediates,
-            datasetTypes.outputs,
+
+        for datasetTypes, is_input, is_output in (
+            (datasetTypes.initIntermediates, True, True),
+            (datasetTypes.initOutputs, False, True),
+            (datasetTypes.intermediates, True, True),
+            (datasetTypes.outputs, False, True),
         ):
```

Comment:
`is_output` is always `True` here, maybe remove it from loop variables? Or did you also want to check pure inputs here?

Reply:
Incredibly, I had not spotted that the `is_output` flag was always true. I had assumed when writing `_register_dataset_types` that there would be cases where input was true but outputs was false but seemingly I'm not making use of that. I suppose I should simplify the code below.